symbolic-debuginfo/src/pe.rs

```diff
@@ -276,6 +278,60 @@ impl<'data> PeObject<'data> {
             self.pe.exception_data.as_ref()
         }
     }
+
+    /// Returns the raw buffer of Embedded Portable PDB Debug directory entry, if any.
+    pub fn embedded_ppdb(&self) -> Result<Option<PeEmbeddedPortablePDB<'data>>, PeError> {
+        // Note: This is currently not supported by goblin, see https://github.com/m4b/goblin/issues/314
+        let opt_header = match self.pe.header.optional_header {
+            Some(v) => v,
+            None => return Ok(None),
+        };
+        let debug_directory = match opt_header.data_directories.get_debug_table().as_ref() {
+            Some(v) => v,
+            None => return Ok(None),
+        };
+        let file_alignment = opt_header.windows_fields.file_alignment;
+        let parse_options = &pe::options::ParseOptions::default();
+        let offset = match pe::utils::find_offset(
+            debug_directory.virtual_address as usize,
+            &self.pe.sections,
+            file_alignment,
+            parse_options,
+        ) {
+            Some(v) => v,
+            None => return Ok(None),
+        };
+
+        use pe::debug::ImageDebugDirectory;
+        let entries = debug_directory.size as usize / std::mem::size_of::<ImageDebugDirectory>();
+        for i in 0..entries {
+            let entry = offset + i * std::mem::size_of::<ImageDebugDirectory>();
+            let idd: ImageDebugDirectory = self.data.pread_with(entry, LE).map_err(PeError::new)?;
+
+            // We're only looking for Embedded Portable PDB Debug Directory Entry (type 17).
+            if idd.data_type == 17 {
+                // ImageDebugDirectory.pointer_to_raw_data stores a raw offset -- not a virtual offset -- which we can use directly
+                let mut offset: usize = match parse_options.resolve_rva {
+                    true => idd.pointer_to_raw_data as usize,
+                    false => idd.address_of_raw_data as usize,
+                };
+
+                // See data specification:
+                // https://github.com/dotnet/runtime/blob/97ddb55e3adde20ceac579d935cef83cfe996169/docs/design/specs/PE-COFF.md#embedded-portable-pdb-debug-directory-entry-type-17
+                offset += 4; // skip `signature`
```

Comment:
You should still check that it matches though :-)

Reply:
that it matches _what_? It doesn't match the pe.header.signature...